libs/s25main/mapGenerator/Textures.cpp

```diff
@@ -0,0 +1,266 @@
+// Copyright (c) 2017 - 2020 Settlers Freaks (sf-team at siedler25.org)
+//
+// This file is part of Return To The Roots.
+//
+// Return To The Roots is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 2 of the License, or
+// (at your option) any later version.
+//
+// Return To The Roots is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Return To The Roots. If not, see <http://www.gnu.org/licenses/>.
+
+#include "mapGenerator/Textures.h"
+#include "mapGenerator/Algorithms.h"
+#include "mapGenerator/TextureHelper.h"
+
+#include <algorithm>
+#include <set>
+#include <stdexcept>
+
+namespace rttr { namespace mapGenerator {
+
+    void TextureMap::Set(const MapPoint& pt, const DescIdx<TerrainDesc>& texture)
```

Comment:
You can take DescIdx by instance/copy. It is a simple number on purpose :)

Reply:
Ah okay, that makes sense. Just replaced all `const DescIdx<TerrainDesc>&` by `DescIdx<TerrainDesc>`.